spec.bs

```diff
@@ -491,12 +494,236 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
+
+<h3 id=parsing-patterns>Parsing Patterns</h3>
+
+<h4 id=tokens>Tokens</h4>
+
+A <dfn>token list</dfn> is a [=list=] containing zero or more [=token=] [=structs=].
+
+A <dfn>token</dfn> is a [=struct=] representing a single lexical token within a [=/pattern string=].
+
+A [=token=] has an associated <dfn for=token>type</dfn>, a string, initially "<a for=token/type>`invalid-char`</a>".  It must be one of the following:
+
+<dl>
+  <dt>"<dfn for=token/type>`open`</dfn>"</dt>
+  <dd>The [=token=] represents a U+007B (`{`) code point.
+  <dt>"<dfn for=token/type>`close`</dfn>"</dt>
+  <dd>The [=token=] represents a U+007D (`}`) code point.
+  <dt>"<dfn for=token/type>`regex`</dfn>"</dt>
+  <dd>The [=token=] represents a string of the form "`(<regular expression>)`".  The regular expression is required to consist of only ASCII code points.
+  <dt>"<dfn for=token/type>`name`</dfn>"</dt>
+  <dd>The [=token=] represents a string of the form "`:<name>`".  The name value is restricted to code points that are consistent with JavaScript identifiers.
+  <dt>"<dfn for=token/type>`char`</dfn>"</dt>
+  <dd>The [=token=] represents a valid pattern code point without any special syntactical meaning.
+  <dt>"<dfn for=token/type>`escaped-char`</dfn>"</dt>
+  <dd>The [=token=] represents a code point escaped using a backslash like "`\<char>`".
+  <dt>"<dfn for=token/type>`other-modifier`</dfn>"</dt>
+  <dd>The [=token=] represents a matching group modifier that is either the U+003F (`?`) or U+002B (`+`) code points.
+  <dt>"<dfn for=token/type>`asterisk`</dfn>"</dt>
+  <dd>The [=token=] represents a U+002A (`*`) code point that can be either a wildcard matching group or a matching group modifier.
+  <dt>"<dfn for=token/type>`end`</dfn>"</dt>
+  <dd>The [=token=] represents the end of the [=/pattern string=].
+  <dt>"<dfn for=token/type>`invalid-char`</dfn>"</dt>
+  <dd>The [=token=] represents a code point that is invalid in the pattern.  This could be because of the code point value itself or due to its location within the pattern relative to other syntactic elements.
+</dl>
+
+A [=token=] has an associated <dfn for=token>index</dfn>, a number, initially 0.  It is the position of the first code point in the [=/pattern string=] represented by the [=token=].
+
+A [=token=] has an associated <dfn for=token>value</dfn>, a string, initially the empty string.  It contains the code points from the [=/pattern string=] represented by the [=token=].
+
+<h4 id=tokenizing>Tokenizing</h4>
+
+A <dfn>tokenize policy</dfn> is a string that must be either "<dfn for="tokenize policy">`strict`</dfn>" or "<dfn for="tokenize policy">`lenient`</dfn>".
+
+A <dfn>tokenizer</dfn> is a [=struct=].
+
+A [=tokenizer=] has an associated <dfn for=tokenizer>input</dfn>, a [=/pattern string=], initially the empty string.
+
+A [=tokenizer=] has an associated <dfn for=tokenizer>policy</dfn>, a [=tokenize policy=], initially "[=tokenize policy/strict=]".
```

Comment:
Here and in two other instances (one strict one lenient) you can make these code font.